composer/callbacks/memory_monitor.py

```diff
@@ -76,24 +76,25 @@ class MemoryMonitor(Callback):
         Memory usage monitoring is only supported for the GPU devices.
     """
 
-    def __init__(self):
-        super().__init__()
-        log.info(
-            "Memory monitor just profiles the current GPU assuming that the memory footprint across GPUs is balanced.")
-        if torch.cuda.device_count() == 0:
-            log.warning("Memory monitor only works on GPU devices.")
+    def fit_start(self, state: State, logger: Logger) -> None:
+        # TODO(ravi) move this check would be on Event.INIT after #1084 is merged
+        # Not relying on `torch.cuda.is_available()` since the model could be on CPU.
+        model_device = next(state.model.parameters()).device
+
+        if model_device.type != 'cuda':
+            raise RuntimeError(
```

Comment:
From @A-Jacobson: 

> I don't like throwing an error here. Now a user will specifically have to delete this callback or their run still stop if they use the same yaml or script on cpu. I'd prefer to disable it for them in that case rather than forcing code/config chances.

Reply:
Sounds good; I will revert to a warning.

Reply:
Made into a warning.